ROADMAP.md

```diff
@@ -0,0 +1,75 @@
+# libp2p daemon roadmap
+
+Revision: r1; 2018-09-25
+
+Authors (a-z):
+  - @bigs
+  - @raulk
+  - @stebalien
+  - @vyzo
+
+## Introduction
+
+The *libp2p daemon* provides a standalone deployment of a libp2p host, running
+in its own OS process and installing a set of virtual endpoints for co-local
+applications to interact with it. The daemon is written in Go and can therefore
+leverage the vast ecosystem of go-libp2p modules.
+
+By running a *single instance* of the daemon per machine, co-local applications
```

Comment:
it's not necessary to run a single instance per box. We can have applications starting their own daemon

Reply:
Agree it's not _necessary_, but what I had in mind was a single daemon-per-machine with a systemd activation-type thing. The first app to ask for a libp2p daemon launches it, and subsequent ones bind to the singleton unix socket.

Multiple daemons don't share connections (unless we implement a clustered design where local daemons gossip about state, and share resources between each other – but that's too complex at this stage), so two daemons == 2x DHT traffic, 2x gossipsub, 2x traffic, duplicate connections to the same peer, etc.

Also, supervising usage of limited resources (e.g. file descriptors, ports, bandwidth, etc.) becomes inefficient. With a single daemon, you could configure quotas for apps, throttle chattier ones, etc. easily. With multiple daemons, we'd require a supervisor process that coordinates daemons, and IMHO it complicates the design going forward.

Reply:
Agreed, I am not arguing against system-wide libp2pd. But its incredible useful for testing to have multiple launched p2pds.

Reply:
Adding:

> Running *multiple instances* of the daemon is also possible, and specially useful for testing purposes.

LGTY?

Reply:
@bigs I'm not convinced we'd need to isolate peerstores, but the daemon would definitely have to manage multiple forward identities. Applications could establish a "session" with the daemon under an identity, and the daemon would keep custody of their keys. I wonder if this is achievable today with go-libp2p, e.g. instantiating multiple hosts with the same peerstore.

I think this is where @Stebalien was going with this point in #3:

> Multi-Tenent from day 1: We'd like much of this daemon to eventually move to the kernel (or, at least, a system daemon) and it turns out that adding in multi-user support later is rather tricky.